azure/src/createAzureClient.ts

```diff
@@ -208,3 +210,29 @@ class AddEndpointPolicy implements PipelinePolicy {
         return await next(request);
     }
 }
+
+/**
+ * The Azure SDK will only throw errors for bad status codes if it has an "operationSpec", but none of our "generic" requests will have that
+ */
+class StatusCodePolicy implements PipelinePolicy {
```

Comment:
This is basically what we had before for the following reason:
> The Azure SDK will only throw errors for bad status codes if it has an "operationSpec", but none of our "generic" requests will have that

OperationSpec isn't something that you can pass into the request anymore, but I believe it's built internally into the SDK. If the SDK call errors out, it throws the error during the pipeline. However, with generic requests, it will accept any status code as a resolved response, so we need to be checking the error and throwing it ourselves.